### PR TITLE
[Feature] Multi-data Source Support for Integrations

### PR DESCRIPTION
**Pull request by sumukhswamy**
### Description
MDS support as part of NEO. https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5870

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/integrations/home.tsx [hunk 1]
```diff
@@ -65,7 +81,13 @@ export const Home = (props: HomeProps) => {
             exact
             path={'/available/:id/setup'}
             render={(routerProps) => (
-              <SetupIntegrationPage integration={decodeURIComponent(routerProps.match.params.id)} />
+              <SetupIntegrationPage
+                integration={decodeURIComponent(routerProps.match.params.id)}
+                dataSourceManagement={dataSourceManagement}
+                notifications={notifications}
```
Comment (ps48) [inline public/components/integrations/home.tsx:87]: Do we need notifications here? Is it used anywhere?

File: public/components/integrations/components/setup_integration_inputs.tsx [hunk 2]
```diff
@@ -190,15 +192,32 @@ export function IntegrationConnectionInputs({
...
+  const onSelectedDataSource = (e) => {
+    const dataConnectionId = e[0] ? e[0].id : undefined;
+    const dataConnectionLabel = e[0] ? e[0].label : undefined;
+    console.log({ dataConnectionId, dataConnectionLabel });
+  };
+  let DataSourceSelector;
+  if (dataSourceEnabled) {
+    DataSourceSelector = dataSourceManagement.ui.DataSourceSelector;
+  }
+  console.log(DataSourceSelector, 'here');
```
Comment (ps48) [inline public/components/integrations/components/setup_integration_inputs.tsx:220]: Let's please remove the `console.logs` from here.

File: public/components/integrations/home.tsx (hunk 1 above)
Comment (sumukhswamy) [reply public/components/integrations/home.tsx:87]: its used for the selector https://github.com/opensearch-project/dashboards-observability/pull/1992/files#diff-8aaf0df3a5da90db7fa5fdde34e79bb6df48fdfce0f0763272ee598266b76c2cR245

**Review by ps48 — APPROVED** [review]

File: public/components/integrations/components/added_integration_table.tsx [hunk 3]
```diff
@@ -45,6 +45,15 @@ export function AddedIntegrationsTable(props: AddedIntegrationsTableProps) {
...
     },
+    {
+      field: 'dataSourceName',
+      name: 'Date Source Name',
+      sortable: true,
+      truncateText: true,
+      render: (value, record) => (
+        <EuiText data-test-subj={`${record.templateName}IntegrationDescription`}>- -</EuiText>
+      ),
+    },
```
Comment (ps48) [inline public/components/integrations/components/added_integration_table.tsx:56]: Should this be under feature flag DataSource Enabled?

**Review by paulstn — APPROVED** [review]

**Pull request merged**